rest_framework/throttling.py

```diff
@@ -155,9 +162,12 @@ def wait(self):
         else:
             remaining_duration = self.duration
 
-        available_requests = self.num_requests - len(self.history) + 1
-        if available_requests <= 0:
-            return None
+        # If we go over the num of requests in history, ensure the
+        # 'available_requests' will stay at 1, suggesting clients to wait for
+        # the full duration of the throttle.
+        available_requests = (
+            self.num_requests - min(self.num_requests, len(self.history)) + 1
+        )
```

Comment:
Probably treat this as a seperate pull request.

Reply:
I have to keep it in, otherwise the suggested duration can be completely wrong - the way it was calculated before didn't work with the new behavior of including failed requests in the history, we can go over `num_requests`.